dandi/tests/test_models.py

```diff
@@ -154,3 +166,265 @@ def test_autogenerated_titles():
     assert schema["title"] == "Asset Meta"
     assert schema["properties"]["schemaVersion"]["title"] == "Schema Version"
     assert schema["definitions"]["PropertyValue"]["title"] == "Property Value"
+
+
+def datacite_post(datacite, doi):
+    """
+    posting the datacite object and returning the status codes
+    of requests.post and requests.get
+    """
+
+    # removing doi in case it exists
+    _clean_doi(doi)
+
+    # checking f I'm able to create doi
+    rp = requests.post(
+        "https://api.test.datacite.org/dois",
+        json=datacite,
+        headers={"Content-Type": "application/vnd.api+json"},
+        auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
+    )
+
+    # checking if i'm able to get the url
+    rg = requests.get(url=f"https://api.test.datacite.org/dois/{doi}/activities")
+
+    # cleaning url
+    _clean_doi(doi)
+    return rp.status_code, rg.status_code
+
+
+def _clean_doi(doi):
+    """removing doi, ignoring the status code"""
+    requests.delete(
+        f"https://api.test.datacite.org/dois/{doi}",
+        auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
+    )
+
+
+@pytest.fixture(scope="module")
+def schema():
+    sr = requests.get(
+        "https://raw.githubusercontent.com/datacite/schema/master/source/"
+        "json/kernel-4.3/datacite_4.3_schema.json"
+    )
+    schema = sr.json()
+    return schema
+
+
+@pytest.fixture(scope="module")
+def client():
+    return DandiAPIClient("https://api.dandiarchive.org/api/")
+
+
+@pytest.mark.parametrize("dandi_id", ["000004", "000008"])
+def test_datacite(dandi_id, schema, client):
```

Comment:
This test should be skipped if `DATACITE_DEV_PASSWORD` is not set so that the test suite can be run without needing that secret.